src/shogun/machine/GaussianProcess.cpp

```diff
@@ -46,6 +46,13 @@ void GaussianProcess::init()
 	SG_ADD(
 	    &m_inducing_features, "inducing_features",
 	    "inducing features for approximation", ParameterProperties::MODEL);
+	add_callback_function("seed", [&]() {
```

Comment:
btw isnt this passing the seed on to all set members already done in the base class? (I thought so at least)

Reply:
you mean already done in `RandomMixin`? it seems like that `RandomMixin` doesn't pass the seed to all set members, as `RandomMixin` doesn't know the members of `GaussianProcess`.

Reply:
Isn't that what `random_seed_callback` does?

Reply:
Maybe I am confused. I always thought the seeding is done recursively to all members no matter what .... but ok

Reply:
again my question here: Didnt we always pass on the seed to all member classes in SGObject?

If not, I think we should. It doesnt seem to make sense that the class (GaussianProcess here) decided to whom to forward the seed right?

Reply:
Yes we do. I think this change has been reverted right?